sound/soc/sof/topology.c

```diff
@@ -1335,7 +1336,12 @@ int sof_pipeline_core_enable(struct snd_sof_dev *sdev,
 			return -ENOENT;
 	}
 
-	return sof_core_enable(sdev, pipeline->core);
+	/* First enable the pipeline core */
+	ret = sof_core_enable(sdev, pipeline->core);
+	if (ret < 0)
+		return ret;
+
+	return sof_core_enable(sdev, swidget->core);
```

Comment:
@lyakh Hmm, the commit message does not seem to cover/mention why you also need changes here?

Reply:
This patch should be merged with the original one, so, this commit message will just disappear. And to better understand this patch one actually should look at both of them merged. The idea behind this change is, that, say, if you get a widget, that `.core == 1` but its pipeline has `.core == 2`. This is a rather weird case and we might have to re-tune it when we actually begin getting such pipelines. Here I'm trying to do at least a safe bet: enable both. Enable is cheap when the core is already enabled, it's then a NOP. But if we really get such a widget and none of the cores have been enabled yet, I decide to first enable the pipeline core, even though actually I don't know what's better or even if there is definitely an option, that's better than the other one.

Reply:
@lyakh Ack, ok. I got confused by the fact this patch is not formatted "fixup!" as "squash!", so thus was expecting a standalone commit message. But hmm, you do have the "Fixes:" tag, so that should be sufficient for maintainers to do the squash. For future, I'd prefer "fixup!" and "squash!" as there is no chance to send those accidently out.